numpy/core/einsumfunc.py

```diff
@@ -1358,11 +1358,27 @@ def einsum(*operands, out=None, optimize=False, **kwargs):
         raise TypeError("Did not understand the following kwargs: %s"
                         % unknown_kwargs)
 
-
     # Build the contraction list and operand
     operands, contraction_list = einsum_path(*operands, optimize=optimize,
                                              einsum_call=True)
 
+    # Handle order kwarg for output array, c_einsum allows mixed case
+    original_order = kwargs.pop('order', 'K')
+    order = original_order.upper()
+    if order not in ['C', 'F', 'A', 'K']:
+        raise ValueError(f"order must be one of 'C', 'F', 'A', or 'K' (got '{original_order}')")
+
+    if order == 'A':
```

Comment:
Thanks, I am curious, maybe we can just make this: `if order == 'A" or order == 'a':` and call `asanyarray` unconditionally? The overhead added by `asanarray` should not be much larger than all the additional logic or the lambda call (it might be, but if that is so, it that is rather a reason to optimize the asanyarray call).

The `asanyarray` should be a `no-op` already if order is `K`.

Reply:
I looked into that, but according to the docs order must be either C or F, but defaults to C. I read it as it will force a copy to C order and K is not supported.

Reply:
The documentation is incorrect, this just forwards to `np.array` with some defaults (which you could also just call directly here, since we need kwargs anyway).  And `np.array` documents all of them. (Unlike einsum `A` for `np.array` actually lets anything pass, I guess it makes sense since `K` was added at some point later).

Reply:
Ah, great to know. Pushed up a commit that switches this out.

I can take the order check out as well, but it might be good to catch this ahead of time and this keeps the error message identical to that of c_einsum.